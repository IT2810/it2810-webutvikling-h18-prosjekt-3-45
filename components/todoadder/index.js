import React, { Component } from 'react';
import { connect } from 'react-redux';
import { View, Modal, StyleSheet } from 'react-native';
import TodoSchema from '../todoschema';
import { addTodo } from '../../features/todos/actions';
import { Icon } from 'expo';

/**
 * Modal functionality is inspired from
 * https://facebook.github.io/react-native/docs/modal.html
 */
export class TodoAdder extends Component {
  state = {
    modalVisible: false,
  };

  /**
   * Sets the visibility of the form modal to the visible-state
   * @param visible   boolean (whether or not the modal should be visible)
   */
  setModalVisibility = visible => {
    this.setState({
      modalVisible: visible,
    });
  };

  /**
   * Creates a new todo with forminput.
   * This will display the todoschema s.t. we can get a form modal
   * @param title         string    title for todo
   * @param description   string    description for todo (optional)
   * @param date          string    date for todo (optional)
   */
  createTodo = ({ title, description, date }) => {
    this.props.addTodo(title, description, date);
    this.setModalVisibility(false);
  };

  render() {
    return (
      <View style={styles.container}>
        <Modal
          animationType="slide"
          transparent={false}
          visible={this.state.modalVisible}
          onRequestClose={() => this.setModalVisibility(false)}
        >
          <View style={styles.schemaContainer}>
            <TodoSchema
              saveForm={this.createTodo}
              hideModule={this.setModalVisibility}
            />
          </View>
        </Modal>

        <View style={styles.btnContainer}>
          <Icon.MaterialIcons
            name="add-circle"
            size={70}
            color={'#00AAEE'}
            onPress={() => {
              this.setModalVisibility(true);
            }}
            underlayColor={'#ffff00'}
          />
        </View>
      </View>
    );
  }
}

const mapDispatchToProps = {
  addTodo,
};

export default connect(
  undefined,
  mapDispatchToProps,
)(TodoAdder);

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },

  schemaContainer: {
    flex: 1,
    paddingVertical: 30,
  },

  btnContainer: {
    marginRight: 20,
    marginBottom: 20,
  },
});
