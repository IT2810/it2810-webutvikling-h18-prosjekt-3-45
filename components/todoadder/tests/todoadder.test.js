import React from 'react';
import renderer from 'react-test-renderer';
import { TodoAdder } from '../../todoadder';

jest.mock('../../todoschema', () => () => 'Schema');

describe('TodoAdder', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TodoAdder />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
