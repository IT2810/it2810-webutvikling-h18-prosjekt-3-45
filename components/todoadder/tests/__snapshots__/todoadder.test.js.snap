// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoAdder renders correctly 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 30,
        }
      }
    >
      TodoSchema
    </View>
  </Modal>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;
